### PR TITLE
Add acceptance testing

### PR DESCRIPTION
**Pull request by tristil**
This adds an acceptance test suite. I'll try to add a login spec soon to demonstrate how it works a little better but this is enough to get started.

Protractor is all about Promises, which I don't really understand but in practice it means using the Chai syntax but always adding `.eventually`. It is also tightly integrated into Angular, so you can do things like make expectations about data bindings.

There are two commits here that aren't really about the main theme but they came up while I was working on this and I didn't want to bother with separate PRs. About the Mongoose change, you will probably want to start using [npm shrinkwrap](https://docs.npmjs.com/cli/shrinkwrap) to avoid having random incompatibilities like this crop up. We just ran into this problem at work, and it makes me wonder why this isn't the standard advice for Node projects.

Also I'll add a unit test suite in another PR.

### CODE REVIEW

File: spec/acceptance/spec_helper.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		fs = require('fs'),
+		app = require('../../server'),
+		debug = require('debug')('gokibitz'),
+		io = require('../../server/io'),
+		server;
+
+chai.use(chaiAsPromised);
+global.expect = expect;
+
+app.set('port', 3435); // Use a different port from development server
```
Comment (neagle) [inline spec/acceptance/spec_helper.js]: Good port number: the 3-4 / 3-5 opening is an advanced but good opening. http://eidogo.com/#18Ie8mufo:0,4

Comment (neagle) [inline spec/acceptance/spec_helper.js]: Good port number: the 3-4 / 3-5 opening is an advanced but good opening. http://eidogo.com/#18Ie8mufo:0,4

File: README.md [hunk 2]
```diff
@@ -104,3 +104,56 @@ If you're working on a feature, feel free to work on it and then unveil it to se
...
+For examples of how to write acceptance tests look in the
+[spec/acceptance](spec/acceptance) folder. You can also look at the
+documentation for [Protractor](http://angular.github.io/protractor/#/) and
+[Chai](http://chaijs.com/) for information about the syntax.
+
+#### Unit tests
+
+These haven't been implemented yet.
+
+### Tmuxinator
```
Comment (neagle) [inline README.md]: I have to admit I actually abandoned Tmuxinator a little while ago: I found I changed my workflow too much as I was going. Now I rely mostly on tmux-resurrect to save/restore my sessions. But in terms of making a nice project workflow available for anyone, this is awesome.

File: README.md [hunk 3]
```diff
@@ -104,3 +104,56 @@ If you're working on a feature, feel free to work on it and then unveil it to se
 ### Error: listen EADDRINUSE
 
 If you're like me and you close your laptop without shutting down your `gulp watch` process, sometimes when you come back you'll close your gulp watch, but you'll still be able to access the site locally and if you try to start up `gulp watch` again, you'll get an `EADDRESSINUSE` error. (And you'll still get console output about requests.) Just type `killall node` on a Mac (Command for Windows?) to shut down the node process and you should be able to start `gulp watch` again. If anyone wants to let me know how to fix this, I'd be much obliged.
+
+### Running Tests
+
+There are two kinds of tests in this project: acceptance tests (sometimes
+called functional or end-to-end tests) and unit tests. Acceptance tests make
+sure that the system is working as a whole and unit tests make sure that the
+logic of individual functions is correct.
```
Comment (neagle) [inline README.md]: This is great documentation.

File: README.md [hunk 4]
```diff
@@ -104,3 +104,56 @@ If you're working on a feature, feel free to work on it and then unveil it to se
...
+```bash
+./bin/protractor spec/acceptance/conf.js
+```
+
+Note: You do not need to be running a development server, but you do need to
+have MongoDB running. The acceptance test starts up its own test app server
+that needs to connect to Mongo.
+
+If all goes well you should see a Chrome window open briefly and you should see
+information about passing (and failing) tests in your terminal.
```
Comment (neagle) [inline README.md]: This worked perfectly for me, following your instructions.

Comment (tristil) [inline README.md]: Phew!

File: README.md (hunk 2 above)
Comment (tristil) [inline README.md]: I just discovered that plugin. I'll have to play around more with it.

**Comment by neagle** [comment]
I think there are two follow-up items for me, now:

* I'd like to get this set up with TravisCI so that it runs tests whenever the project gets pushed.
* I need to set up a gulp task to run the acceptance tests.

**Comment by tristil** [comment]
:+1: I'll add checkboxes to the issue I opened to keep track of the tasks

**Pull request merged**